scalahost/nsc/src/main/scala/scala/meta/internal/semantic/LengthWriter.scala

```diff
@@ -0,0 +1,17 @@
+package scala.meta.internal.semantic
+
+import java.io.Writer
+
+// A writer that keeps track of the current length.
+class LengthWriter(delegate: Writer, start: Int) extends Writer {
+  var length: Int = start
+  override def flush(): Unit = {
+    length = 0
+    delegate.flush()
```

Comment:
`length = 0` does not make any sense here. It should stay the same. `flush()` does not "clear" the contents of the writer; it only ensures all written data are pushed down to the final sink, without buffering.